src/controllers/gist.ts

```diff
@@ -98,6 +109,44 @@ class GistController implements BaseController {
     }
   }
 
+  /**
+   * Route to create a gist.
+   *
+   * @param {Request} req Request object
+   * @param {Response} res Response object
+   */
+  private createGist = (req, res) => {
+    if (req.user === undefined) {
+      res.redirect(redirectUrl.failure)
+    }
+    else {
+      const oauthToken = req.user.accessToken
+      const body = req.body as ICreateGist
+      const { content, name, privacy, title='' } = body
+      fetch(`https://api.github.com/gists?oauth_token=${oauthToken}`, {
+        method: 'post',
+        header: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          description: title,
+          public: !privacy,
+          files: {
+            [`${name}.json`]: {
+              content,
+            },
+          },
+        }),
+      })
+        .then(res => res.json())
```

Comment:
It is not required. There are two ways of dealing with fetch which are promises and using async/await. I have chosen the former because it looks cleaner to me.